spec/controllers/hq/sponsors_controller_spec.rb

```diff
@@ -30,38 +34,36 @@
 
   context '#create' do
     specify 'successful' do
-      expect(Sponsor).to receive(:new).and_return(@sponsor_new)
-      expect {post :create, sponsor: @sponsor_build.attributes }.to change{Sponsor.count}.by 1
-      expect(response).to redirect_to hq_sponsor_url(Sponsor.find_by_name(@sponsor_build.name))
+      post :create, sponsor: build(:sponsor).attributes
+      expect(response).to redirect_to hq_sponsor_url(assigns :sponsor)
     end
```

Comment:
Why I used `build(:sponsor).attributes`.
Because an `agent` is needed for saving the new sponsor.

```ruby
ss = Sponsor.new( FactoryGirl.build(:sponsor).attributes ); ss.save    ->  true

ss = Sponsor.new( FactoryGirl.build_stubbed(:sponsor).attributes ); ss.save    ->  false
ss.errors   ->    @messages={:agent=>["can't be blank"]}>

ss = Sponsor.new( FactoryGirl.attributes_for(:sponsor) ); ss.save      -> false
ss.errors   ->    @messages={:agent=>["can't be blank"]}>
```